server/data_server.c

```diff
@@ -127,11 +130,22 @@ static void capture_experiment(void)
             count = hw_read_streamed_data(block, DATA_BLOCK_SIZE, &at_eof);
         while (data_thread_running  &&  count == 0  &&  !at_eof);
         if (count > 0)
+        {
+            if (!ts_captured)
+            {
+                hw_get_start_ts(&pcap_start_ts);
+                ts_captured = true;
+            }
             release_write_block(data_buffer, count);
+        }
 
         total_bytes += count;
         experiment_sample_count = total_bytes / sample_length;
     }
+
+    if (!ts_captured)
+            release_write_block(data_buffer, 0);
```

Comment:
Interesting.  If this is needed it's addressing an outstanding bug, so I'd expect the condition should be `count == 0`, as a missing `release_write_block` will happen any time the capture end on a timeout.  Also, there is an indentation error here!

Maybe `end_write` is papering over the cracks here and triggering the missing `release_write_block`, because we haven't seen a problem with this before ... or have we?